lib/resty/t1k.lua

```diff
@@ -52,7 +65,22 @@ function _M.do_access(opts)
         opts.remote_addr_idx = idx
     end
 
-    request.do_request(opts)
+    local ok, err, result = request.do_request(opts)
+    if not ok then
+        nlog(err_fmt("failed to do request: %s", err))
+        return
+    end
+
+    if mode == consts.MODE_MONITOR then
+        nlog(debug_fmt("monitor mode, skip blocking"))
+        return
+    end
+
+    ok, err = handler.handle(result)
+    if not ok then
+        nlog(err_fmt("failed to handle result: %s", err))
+        return
+    end
```

Comment:
This is the same as before, could you just return the result without performing the interception action?

Reply:
pls take a second look